### ISSUE TITLE
Twitter Redux - Week #4 assignment

### ISSUE THREAD

**Issue by nandaja** [nandaja/androiddummy#5]
@codepathreview 
@codepath

My Simple twitter client app (Refer https://github.com/nandaja/androiddummy/tree/master/MYTwitterApp) is complete with all required & all the suggested features, except for sending direct messages 

I had a couple of queries  : 

1) Is it recommended to execute with just one activity and multiple fragments ? I had to duplicate some of the action ( retweet/ favorite etc ) code in multiple activites. If we are working off of one activity, this could have been solved. 

2)  I tried to work off of a single activity - enclosing the tab slider / view pager in a fragment using a frame layout for the home / mentions timelines & modeling the profile fragment similarly - I could get this working within a frame layout - however, upon hitting back button data would not reload . 

3) How do we customize the text on the pager sliding tab strip , as  in show multiple lines of text etc , wrap around text etc - primarily customize the tab title ? Can we substitute a view in place of the CharSequence which is the title currently ?  

4) The profile images from twitter seem very blurred upon loading into image view.  I attempted resize on Picasso as well - did not seem to help much.

**Comment by codepathreview** [comment]
:+1: Project looks good. Great work implementing so many optionals and working on UI enhancements, the app looks very polished. The way you have architected the fragment looks good, but a few functionalities can still be abstracted out, good job otherwise.

I have provided a detailed [Project 4 Feedback Guide here](http://courses.codepath.com/snippets/intro_to_android/project_4_feedback) which covers the most common issues with this submitted project. Read through the feedback guide point-by-point to determine how you might be able to improve your submission.

Let us know if you have any other thoughts or questions about this assignment. Hopefully you can see this coming together as a "fully fledged" twitter client with some more work and polish. This app contains all of the components now (fragments, models, networking, client, tab navigation, image loading, et al) of 90% of dynamic data-driven API client. Obviously there are lots of details and patterns to learn, but by this point you have been introduced to all the major frameworks and concepts. Hopefully you would feel fairly confident getting started making Android apps for instagram, pinterest, yardsale, flickr, <insert api client here> using the same patterns.

**Comment by nesquena** [comment]
> 1) Is it recommended to execute with just one activity and multiple fragments ? I had to duplicate some of the action ( retweet/ favorite etc ) code in multiple activites. If we are working off of one activity, this could have been solved.

You can have multiple fragments and one activity, in some cases it can make things easier and in others it can make the Activity very messy and hard to follow. It's important to try and strike the right balance. Even if you have multiple activities, it can be useful to re-use fragments to share logic. 

> 3) How do we customize the text on the pager sliding tab strip , as in show multiple lines of text etc , wrap around text etc - primarily customize the tab title ? Can we substitute a view in place of the CharSequence which is the title currently ?

The PagerSlidingTabStrip is not particularly flexible in that way. For more flexibility, you can opt to essentially rebuild the tab indicator yourself [using this guide](https://github.com/codepath/android_guides/wiki/Google-Play-Style-Tabs-using-SlidingTabLayout) which requires a bit more effort (not too much) and then you can customize the tab view to look however you want.

> 4) The profile images from twitter seem very blurred upon loading into image view. I attempted resize on Picasso as well - did not seem to help much

Ah good question, yes. See Tip 10 [on helpful hints](http://courses.codepath.com/courses/intro_to_android/week/3#!hints)
